tests/functional/aws-node-sdk/test/multipleBackend/acl/aclAwsVersioning.js

```diff
@@ -0,0 +1,228 @@
+
+function putObjectAcl(s3, key, versionId, acp, cb) {
+    s3.putObjectAcl({ Bucket: bucket, Key: key, AccessControlPolicy: acp,
+        VersionId: versionId }, err => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `putting object acl, got error ${err}`);
+        cb();
+    });
+}
+
+function putObjectAndAcl(s3, key, body, acp, cb) {
+    s3.putObject({ Bucket: bucket, Key: key, Body: body,
+    Metadata: { 'scal-location-constraint': awsLocation } },
+    (err, putData) => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `putting object, got error ${err}`);
+        putObjectAcl(s3, key, putData.VersionId, acp, () =>
+            cb(null, putData.VersionId));
+    });
+}
+
+function putVersionsWithAclToAws(s3, key, data, acps, cb) {
+    if (data.length !== acps.length) {
+        throw new Error('length of data and acp arrays must be the same');
+    }
+    enableVersioning(s3, bucket, () => {
+        async.timesLimit(data.length, 1, (i, next) => {
+            putObjectAndAcl(s3, key, data[i], acps[i], next);
+        }, (err, results) => {
+            assert.strictEqual(err, null, 'Expected success ' +
+                `putting versions with acl, got error ${err}`);
+            cb(null, results);
+        });
+    });
+}
+
+function getObjectAndAssertAcl(s3, params, cb) {
+    const { bucket, key, versionId, body, expectedVersionId, expectedResult }
+        = params;
+    getAndAssertResult(s3, { bucket, key, versionId, expectedVersionId, body },
+        () => {
+            s3.getObjectAcl({ Bucket: bucket, Key: key, VersionId: versionId },
+                (err, data) => {
+                    assert.strictEqual(err, null, 'Expected success ' +
+                        `getting object acl, got error ${err}`);
+                    assert.deepEqual(data, expectedResult);
+                    cb();
+                });
+        });
+}
+function getObjectAndAssertAcls(s3, key, versionIds, expectedData, expectedAcps,
+    cb) {
+    async.timesLimit(versionIds.length, 1, (i, next) => {
+        const versionId = versionIds[i];
+        const body = expectedData[i];
+        const expectedResult = expectedAcps[i];
+        getObjectAndAssertAcl(s3, { bucket, key, versionId, body,
+            expectedResult, expectedVersionId: versionId }, next);
+    }, err => {
+        assert.strictEqual(err, null, 'Expected success ' +
+            `getting object acls, got error ${err}`);
+        cb();
+    });
+}
+
+describeSkipIfNotMultiple('AWS backend put/get object acl with versioning',
+function testSuite() {
+    this.timeout(30000);
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+
+        beforeEach(() => {
+            process.stdout.write('Creating bucket');
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
```

Comment:
I'd rather if we were testing the location constraint based on the bucket location.